### PR TITLE
Added Workspace Quick Switcher

### PR DESCRIPTION
**Pull request by vabene1111**
## Reason for PullRequest
Especially with the old Workspace File Dialog, but also with the new one, it was a little annoying to always search the Path of the Workspace you wanted to work in, so i added a List with the last 5 (configurable) workspaces a users has used for quick switching.

## Description of the Changes
Added a List Box to the WorkSpace Selector Dialog that Contains the last 5 used workspaces.
Clicking an Item results in it beeing put in the TextBox above and the "ok" button beeing enabled, pressing ok will then open the selected workspace.
[Screenshot](https://drive.google.com/uc?id=0B_TmB6GeWD7PUmNjb19ObWVJYXc)
## Notes
I have tested the changes with a clean install and also different data and everything seems to work fine.
I have tried to make all changes respecting the wpf mvvm scheme (and the general style of the application), but since i have never used it before there might be mistakes in that, if anything needs to be changed please let me know.

### CODE REVIEW

**Comment by dedmen** [comment]
I don't know if it's the best time to do something like that now. A ton of the workspace stuff will probably change with the new Plugin-System.

File: ArmA.Studio/Properties/Localization.resx [hunk 1]
```diff
@@ -397,6 +397,9 @@ Do you want to update now?</value>
   <data name="WorkspaceSelectorDialog_Browse" xml:space="preserve">
     <value>Select Existing</value>
   </data>
+  <data name="WorkspaceSelectorDialog_SelectPrev" xml:space="preserve">
+    <value>Last Workspaces</value>
```
Comment (X39) [inline ArmA.Studio/Properties/Localization.resx]: Should be "Recent Workspaces" or just "Recent"

File: ArmA.Studio/Dialogs/WorkspaceSelectorDialogDataContext.cs [hunk 2]
```diff
@@ -18,6 +19,12 @@ public class WorkspaceSelectorDialogDataContext : IDialogContext
         public string CurrentPath { get { return this._CurrentPath; } set { this._CurrentPath = value; this.OKButtonEnabled = !string.IsNullOrWhiteSpace(value); this.RaisePropertyChanged(); } }
         private string _CurrentPath;
 
+        public List<String> WorkSpaceListBox { get { return this._WorkSpaceListBox;  } set { this._WorkSpaceListBox = value; this.RaisePropertyChanged(); } }
+        private List<String> _WorkSpaceListBox;
+
+        public String WorkSpaceListBoxSelectedItem { get { return this._WorkSpaceListBoxSelectedItem; } set { this._WorkSpaceListBoxSelectedItem = value; this.RaisePropertyChanged();  this.WorkSpaceListBoxSelectionChanged(); } }
```
Comment (X39) [inline ArmA.Studio/Dialogs/WorkspaceSelectorDialogDataContext.cs]: Extra field and property WorkSpaceListBoxSelectedItem should be removed and the SelectedItem can be bound to `CurrentPath` directly

File: ArmA.Studio/Dialogs/WorkspaceSelectorDialog.xaml [hunk 3]
```diff
@@ -7,14 +7,25 @@
...
             <ColumnDefinition Width="8"/>
             <ColumnDefinition Width="Auto"/>
         </Grid.ColumnDefinitions>
-        <TextBox Grid.Column="0" IsReadOnly="True" Text="{Binding CurrentPath, UpdateSourceTrigger=PropertyChanged}" Foreground="Gray"/>
-        <Button Grid.Column="2" Content="{x:Static p:Localization.WorkspaceSelectorDialog_Browse}" Padding="8 0" MinWidth="100" Command="{Binding CmdBrowse}"/>
+        <TextBox Grid.Row="0" Grid.Column="0" IsReadOnly="True" Text="{Binding CurrentPath, UpdateSourceTrigger=PropertyChanged}" Foreground="Gray"/>
+        <Button Grid.Row="0" Grid.Column="2" Content="{x:Static p:Localization.WorkspaceSelectorDialog_Browse}" Padding="8 0" MinWidth="100" Command="{Binding CmdBrowse}"/>
+
+        <Label Grid.Row="1" Content="{x:Static p:Localization.WorkspaceSelectorDialog_SelectPrev}"></Label>
+        <ListBox Grid.Row="2" Grid.ColumnSpan="3" ItemsSource="{Binding WorkSpaceListBox}" SelectedItem="{Binding WorkSpaceListBoxSelectedItem}" >
```
Comment (X39) [inline ArmA.Studio/Dialogs/WorkspaceSelectorDialog.xaml]: Extra field WorkSpaceListBoxSelectedItem should be removed and the SelectedItem can be bound to `CurrentPath` directly

File: ArmA.Studio/ConfigHost.cs [hunk 4]
```diff
@@ -56,6 +56,27 @@ public static string WorkspacePath
...
+                    }
+                    
+                    return new List<string>(pathArray);
+                }
+                set {
+                    string s = string.Join(",", value.ToArray());
+
+                    Instance.AppIni.SetValue(nameof(App), nameof(PrevWorkspacePath), s);
+                    Instance.Save(EIniSelector.App);
+                }
```
Comment (X39) [inline ArmA.Studio/ConfigHost.cs:79]: the `{`s should be placed below `get`/`set`

File: ArmA.Studio/App.xaml.cs [hunk 5]
```diff
@@ -126,6 +126,21 @@ public static bool SwitchWorkspace()
...
+                prevWorkSpaces.Remove(workspace);
+            }
+            prevWorkSpaces.Insert(0, workspace);
+            int numSpaces = 5;
+            if(prevWorkSpaces.Count > numSpaces)
+            {
+                prevWorkSpaces.RemoveRange(numSpaces, (prevWorkSpaces.Count- numSpaces));
+            }
+            ConfigHost.App.PrevWorkspacePath = prevWorkSpaces;
+
```
Comment (X39) [inline ArmA.Studio/App.xaml.cs]: Should be moved into the OK command in `ArmA.Studio/Dialogs/WorkspaceSelectorDialogDataContext.cs`

**Review by X39 — CHANGES_REQUESTED** [review]

**Comment by vabene1111** [comment]
i will fix @X39 requests tomorrow, if you think its not the best time feel free to just leave it in pending or reject, however you like

**Review by X39 — APPROVED** [review]

**Pull request merged**